SU-HSDO/suhumsci
# Feat(STN-883):  Vertical timeline pattern for views

# READY FOR REVIEW

## Summary
As a Developer, I would like to use the Vertical Timeline pattern in a view and have the styles appear like the pattern used in a paragraph type.

This work aims to address some specific issues that do not currently work as we would like without impacting the current functionality of the Vertical Timeline as a Paragraph.

- **Current issue**: _" The Variants for Closed and Open should function as expected. If necessary, use default behavior (expanded). "_
  - This issue was addressed utilizing a class on the view to modify the functionality.
- **Current issue:** _" Currently the pattern appears without the vertical connecting lines. "_
  -  This issue was addressed using CSS only.


Jira Card: https://sparkbox.atlassian.net/browse/STN-883

## Need Review By (Date)
1/3

## Urgency
medium

## Steps to Test
1. npm run test
2. To check the styling of the Vertical Timeline in a View you can use [this page](https://sparkbox-sandbox.suhumsci.loc/example-vertical-timeline-view) if you have synced with staging recently.
3. You can compare the styling against[ this page](https://sparkbox-sandbox.suhumsci.loc/qa-vertical-timeline-raised-cards) that is also available to sync from staging, this is the paragraph display of the Vertical Timeline.
4. Although the Pattern Variant setting within the view does not work to adjust the collapse and expansion of the timeline items, we have a utility workaround for this functionality.
- If you go to the View page for the Timeline View to [edit it here](https://sparkbox-sandbox.suhumsci.loc/admin/structure/views/view/custm_custom_econ_events/edit/block_11).
- Then add the CSS class `hb-timeline__collapsed` to the View.
![image](https://user-images.githubusercontent.com/9355698/147295910-4d622dfc-ff2a-44a6-8462-3e7cc4fdefde.png)
- Save the View.
- This allows the existing JS for Vertical Timeline to function and allows the Timeline Items to be collapsed by default by correctly adjusting the Aria Expand attributes.
- This should meet the requirements set in the JIRA card for default functionality and allow the use of collapse if needed.

## Verification Checklist:
- [x] The Paragraph usage of Vertical Timeline styling is unaffected, functionality and styling are the same as previous.
- [x] The Vertical Timeline in Views has the ability to be collapsed on load by using the steps to add the `hb-timeline__collapsed` class in the View.
- [x] Using `hb-raised-cards` or `hb-raised-cards--uniform-height` doesn't alter the styling of the Vertical Timeline in a View.
- [x] This works as it should in both Traditional and Colorful themes.
- [x] This works properly with all color pairings, with the connecting lines and dots changing to the correct color pairing.
- [x] Verify the documentation has been updated to track the usage of the `hb-timeline__collapsed` [utility class doc update](https://github.com/SU-HSDO/suhumsci/blob/9b258c3da5cc605b75211c028fc3f142b8096130/docroot/themes/humsci/humsci_basic/docs/utility-classes.md)

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
